Dev fp vs oop/README.md

```diff
@@ -0,0 +1,107 @@
+
+        Output: Squared numbers: [1, 4, 9, 16, 25, 36, 49, 64, 81, 100]
+
+        // Example 2: Using filter to select even numbers
+        List<Integer> evenNumbers = numbers.stream()
+                                           .filter(x -> x % 2 == 0)
+                                           .toList();
+        System.out.println("Even numbers: " + evenNumbers);
+        Output: Even numbers: [2, 4, 6, 8, 10]
+
+        // Example 3: Using reduce to find the sum of all numbers
+        int sumOfNumbers = numbers.stream()
+                                  .reduce(0, Integer::sum);
+        System.out.println("Sum of numbers: " + sumOfNumbers);
+        Output: Sum of numbers: 55
+    }
+}
+
+## Object Oriented design patterns
+Object-oriented design patterns are like templates for solving common problems when making software. They help organize code in a way that makes it easier to understand and work with. Some examples are Singleton, Factory Method, Abstract Factory, Builder, Prototype, Adapter, Decorator, Composite, Proxy, Facade, Observer, Strategy, State, Template Method, and Command patterns. Each pattern helps with different parts of making software, like creating objects, changing their behavior, or making them work together. They're like handy tools that developers can use to build better software more efficiently.
+
+## Example of the same code used in functional programming 
+public class NumberManipulator {
+    private List<Integer> numbers;
+
+    public NumberManipulator(List<Integer> numbers) {
+        this.numbers = numbers;
+    }
+
+    public List<Integer> squareNumbers() {
+        List<Integer> squaredNumbers = new ArrayList<>();
+        for (int num : numbers) {
+            squaredNumbers.add(num * num);
+        }
+        return squaredNumbers;
+    }
+
+    public List<Integer> getEvenNumbers() {
+        List<Integer> evenNumbers = new ArrayList<>();
+        for (int num : numbers) {
+            if (num % 2 == 0) {
+                evenNumbers.add(num);
+            }
+        }
+        return evenNumbers;
+    }
+
+    public int sumNumbers() {
+        int sum = 0;
+        for (int num : numbers) {
+            sum += num;
+        }
+        return sum;
+    }
+
+    public static void main(String[] args) {
+        // Sample list of numbers
+        List<Integer> numbers = List.of(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
+        NumberManipulator manipulator = new NumberManipulator(numbers);
+
+        // Example 1: Square each number
+        List<Integer> squaredNumbers = manipulator.squareNumbers();
+        System.out.println("Squared numbers: " + squaredNumbers);
+
+        // Example 2: Select even numbers
+        List<Integer> evenNumbers = manipulator.getEvenNumbers();
+        System.out.println("Even numbers: " + evenNumbers);
+
+        // Example 3: Sum all numbers
+        int sum = manipulator.sumNumbers();
+        System.out.println("Sum of numbers: " + sum);
+    }
+}
```

Comment:
This is probably more code than is necessary here. You've provided some good functional examples above, but to reproduce all these in OOP seems unnecessary. Perhaps choosing just one (`getEvenNumbers` is a nice contrast with the `filter` example, or perhaps an OOP version of the map-filter-reduce example I've suggested you create), and also removing the `main` method - as it doesn't really add to the example as a whole.